test/e2e/image_registry_test.go

```diff
@@ -2,88 +2,55 @@ package e2e
 
 import (
 	"net/url"
-	"strings"
 	"testing"
 
-	v1 "k8s.io/api/core/v1"
+	"github.com/stretchr/testify/require"
 )
 
+// TestImageRegistryPods ensure that all the containers images in openshift-monitoring
+// are from the same registry than the CMO's image.
 func TestImageRegistryPods(t *testing.T) {
-	var pods *v1.PodList
-
-	// Get all pods in openshift-monitoring namespace.
-	var urlRegistry string
-	pods = f.MustGetPods(t, f.Ns)
-
-	// use CMO image's registry as a reference for all other containers
-	for _, pod := range pods.Items {
-		if strings.Contains(pod.Name, "cluster-monitoring-operator") {
-			imageUrl, err := url.Parse("stubheader://" + pod.Spec.Containers[0].Image)
-			if err != nil {
-				t.Fatalf("Fail to decode host: %v", err)
-			}
-			urlRegistry = imageUrl.Host
-			break
-		}
-	}
+	cmoImageRegistryIsUsedInNsAssert(t, f.Ns)
+}
 
-	if urlRegistry == "" {
-		t.Fatalf("CMO pod not found")
+func cmoImageRegistryIsUsedInNsAssert(t *testing.T, ns string) func(t *testing.T) {
+	return func(t *testing.T) {
+		assertCMOImageRegistryIsUsed(t, ns)
 	}
+}
 
-	for _, pod := range pods.Items {
-
-		for _, container := range pod.Spec.Containers {
-
-			// We consider the hostname part of image URL be the image registry
-			imageUrl, err := url.Parse("stubheader://" + container.Image)
-			if err != nil {
-				t.Fatalf("Fail to decode host: %v", err)
-			}
-
-			if imageUrl.Host != urlRegistry {
-				t.Fatalf("Pod %s Container %s registry %s differs from CMO registry %s", pod.Name, container.Name, imageUrl.Host, urlRegistry)
-			}
+func assertCMOImageRegistryIsUsed(t *testing.T, ns string) {
+	getRegistry := func(t *testing.T, image string) string {
+		// This first attempt is needed; otherwise, we may blindly add a second scheme,
+		// and the initial one will be considered the hostname.
+		u, err := url.ParseRequestURI(image)
+		if err == nil {
+			return u.Host
 		}
-
+		// Maybe no scheme, add one.
+		u, err = url.ParseRequestURI("stubheader://" + image)
+		require.NoError(t, err)
+		return u.Host
 	}
 
-	setupUserWorkloadAssetsWithTeardownHook(t, f)
-	uwmCM := f.BuildUserWorkloadConfigMap(t,
-		`prometheus:
-  enforcedTargetLimit: 10
-  volumeClaimTemplate:
-    spec:
-      resources:
-        requests:
-          storage: 2Gi
-`,
-	)
-	f.MustCreateOrUpdateConfigMap(t, uwmCM)
-	defer f.MustDeleteConfigMap(t, uwmCM)
+	cmoPod := f.MustListPods(t, f.Ns, "app.kubernetes.io/name=cluster-monitoring-operator")
```

Comment:
```suggestion
	cmoPod := f.MustListPods(t, ns, "app.kubernetes.io/name=cluster-monitoring-operator")
```
Was this one missed?